lib/elixir/lib/kernel/typespec.ex

```diff
@@ -250,6 +250,8 @@ defmodule Kernel.Typespec do
 
   defp translate_type({_, {kind, {:::, _, [{name, _, args}, definition]}, pos}}, state) do
     caller = :elixir_locals.get_cached_env(pos)
+    # Clean local state
+    state = %{state | local_vars: %{}}
```

Comment:
Since this is called multiple times, what if we put it in a function called exactly `clean_local_state`?

Reply:
I like that 👍